src/poetry/console/application.py

```diff
@@ -221,7 +221,7 @@ def register_command_loggers(
 
         io = event.io
 
-        loggers = [
+        loggers: list[Any] = [
```

Comment:
Although this silences `mypy`, `loggers` is really only a `list[str]`. The actual issue stems from re-using the same variable name for different types. So a more direct solution would be to avoid it on line 235:

```python
for logger_name in loggers:
    logger = logging.getLogger(logger_name)
```